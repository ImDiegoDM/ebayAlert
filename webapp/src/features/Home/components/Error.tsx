import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  margin: 50px 0;
  text-align: center;
  display: flex;
  flex-direction: column;
`;

interface ErrorProps {
  onTryAgain?: () => void;
}

export function Error(props: ErrorProps) {
  return <Container>
    <span>Opss.. it seems an error has ocurred getting the alerts</span>
    <div>
      <button onClick={props.onTryAgain}> try again </button>
    </div>
  </Container>;
}
