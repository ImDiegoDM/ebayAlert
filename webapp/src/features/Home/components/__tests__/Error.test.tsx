import { render } from '@testing-library/react';
import React from 'react';
import { Error } from '../Error';

describe('<Error/>', () => {
  it('should match snapshot', () => {
    const { container } = render(<Error/>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
