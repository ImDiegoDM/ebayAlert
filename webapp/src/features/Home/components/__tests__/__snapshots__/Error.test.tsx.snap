// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> should match snapshot 1`] = `
.c0 {
  margin: 50px 0;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
>
  <span>
    Opss.. it seems an error has ocurred getting the alerts
  </span>
  <div>
    <button>
       try again 
    </button>
  </div>
</div>
`;
